src/positions.rs

```diff
@@ -279,7 +279,18 @@ impl Position {
     }
 
     pub fn gen_knight_moves(&self, from: Square, m: &masks::Lookup) -> Vec<Move> {
-        self.gen_from_atk(from, m.knight[from as usize])
+        let mut ret: Vec<Move> = Vec::new();
+        for moves in self.gen_from_atk(from, m.knight[from as usize]) {
```

Comment:
```suggestion
        for move in self.gen_from_atk(from, m.knight[from as usize]) {
```

`move` makes more sense because you're iterating over individual moves